src/java.base/share/classes/sun/net/www/protocol/file/FileURLConnection.java

```diff
@@ -90,8 +90,12 @@ public void connect() throws IOException {
         }
     }
 
-    public boolean isConnected() {
-        return connected;
+    public synchronized void closeInputStream() throws IOException {
+        if (is != null) {
+            is.close();
+            is = null;
+            connected = false;
```

Comment:
Not sure about switching `connected` to false or setting `is` to null. That might allow to call `connect` again and might have unintended side effects?

Reply:
Well yes - exactly my point. At the moment if you call getInputStream() twice in succession you will get the same input stream. The transition from not connected to connected is stable. `connect` can only transition once from `false` to `true`.  There's nothing that sets `is` back to `null` or `connect` back to `false` so I believe it should stay that way.

Let's keep the (behaviour) changes minimal.